### DISCUSSION TITLE
more properties for vector3 like vector3.xz

### DISCUSSION THREAD

**Discussion by Nitbundier** [godotengine/godot-proposals#10621] [3D]
right now if you want to operate on 2 axis of a vector3 you have to make a vector2 variable and feed it the vector3 axis you want than operate on that vector2 (or operate seperatly on the axis with out a vector2 but than you cant use the vector's functions) and than feed that vector2 to the vector3.

like in the same code that made me think of this, i want object 'b' to look at object 'a' but only in the y axis: 
```GDScript
extends Node3D
func _process(delta: float) -> void:
	var a : Node3D = %Object1
	var b : Node3D = %Object2

	var angle := Vector2(b.global_position.x,b.global_position.z).angle_to_point(Vector2(a.global_position.x,a.global_position.z))

	b.rotation_degrees.y = rad_to_deg(angle)
```

i was thinking that more propertis like 'xy' that would operate on the x and y axis of the vector3 might make this operation easier.

that would mean 3 new propertis : 'xy' , 'xz' , 'yx'

the same code would translate to:

```GDScript
extends Node3D
func _process(delta: float) -> void:
	var a : Node3D = %Object1
	var b : Node3D = %Object2

	var angle := b.global_position.xz.angle_to_point(a.global_position.xz)

	b.rotation_degrees.y = rad_to_deg(angle)
```

the get function of them just return an vector2 with the corresponding axis of the vector3
and the set function would set the axis of the vector3.

since vector3 is used alot in 3d games this might be a good quality of life feature.

sorry if bad Phrasing , english is my Second language.
i did search for similar discussion but didn't find any so i made this.

**Comment by AThousandShips** [comment]
Thank you for your proposal, see in:
* https://github.com/godotengine/godot-proposals/issues/727
